src/ui_utils.c

```diff
@@ -3217,3 +3217,48 @@ gboolean ui_encodings_combo_box_set_active_encoding(GtkComboBox *combo, gint enc
 	}
 	return FALSE;
 }
+
+GtkWidget* get_search_bar_sbox(GeanyDocument *doc)
+{
+	GtkWidget *page, *vbox;
+	page = GTK_WIDGET(doc->editor->sci);
+	vbox = gtk_widget_get_ancestor(page, GTK_TYPE_VBOX);
+	GList *vbox_children = gtk_container_get_children(GTK_CONTAINER(vbox));
+	GtkWidget *sbox;
+	if (interface_prefs.search_bar_position == SEARCH_BAR_POSITION_TOP)
+	{
+		sbox = (GtkWidget*)vbox_children->data;
+	}
+	else
+	{
+		sbox = (GtkWidget*)((GList*)g_list_last(vbox_children))->data;
+	}
+	g_list_free(vbox_children);
+	return sbox;
+}
+
+void ui_emit_entry_what_to_search_in_search_bar_changed_if_visible(GeanyDocument *doc)
```

Comment:
:astonished:

Not to bikeshed, but it might be useful to spend some of those characters explaining what `sbox` is inside the function. Perhaps it could be shortened to `ui_emit_search_bar_entry_changed`? Likewise for below, it could be something like `ui_emit_search_bar_close_button_clicked`. While I'm a big fan of descriptive function names, it doesn't necessarily have to say everything the function does.

Reply:
ueewtsisbciv() FTW!!! :smiling_imp:

Reply:
@codebrainz Done.

Reply:
And now make it sensible.